### ISSUE TITLE
mpfr_complex allows implicit conversions from integral and real types

### ISSUE THREAD

**Issue by ofloveandhate** [bertiniteam/b2#26]
The provided mpfr complex type allows implicit conversion from integral types, and mpfr_float.  double conversion is disallowed.  

this is currently the case to allow some operators in Eigen to work, because they rely on implicit conversions to double from integral types.  however, for multiple precision types, this is a performance liability, both in terms of the creation of the temporary high-precision number, as well as using complex-complex arithmetic rather than complex-real.  

perhaps in Eigen 3.3, this will be fixed.  if not, we will need to prompt them to fix this, particularly for *= operators, where I originally found the implicit conversion was happening.

in short, please add explicit to constructors for mpfr_complex in Bertini2.

**Comment by ofloveandhate** [comment]
this is not fixed in 3.3.  maybe 3.4?  until eigen fixes this, we probably cannot remove these conversions.

**Comment by ofloveandhate** [comment]
i have asked two questions about this:
* [on QR and SVD](https://forum.kde.org/viewtopic.php?f=74&t=139725) at KDE forum
* [on *=](https://stackoverflow.com/questions/43477416/custom-type-matrix-int-in-eigen) at SO

solution for the two decompositions, but not yet for *=.  so almost there, but not quite.

**Comment by ofloveandhate** [comment]
This is long-solved since I removed the hand-spun mpfr_complex from bertini 2 years ago in favor of boost.multiprecision.
